float64[] min_dists
float64[] max_dists
